src/api/comment.tsx

```diff
@@ -0,0 +1,86 @@
+import { ApiResponse } from "apisauce";
+import api from "./base";
+import {
+  fetchPostCommentsApiPayload,
+  fetchPostCommentsApiResponse,
+  createPostCommentApiResponse,
+  createPostCommentApiPayload,
+  updatePostCommentApiResponse,
+  updatePostCommentApiPayload,
+  deletePostCommentApiResponse,
+  deletePostCommentApiPayload,
+} from "./commentType";
+
+/**
+ * Fetch all comments based on postId
+ */
+const fetchPostComments = ({
+  payload,
+}: fetchPostCommentsApiPayload): Promise<
+  ApiResponse<fetchPostCommentsApiResponse>
+> => {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { postId } = payload;
+  return api.get("/api/v1/posts/${postId}/comments?limit=9999");
+};
+
+/**
+ * Create a comment for a specific post
+ */
+const createPostComment = ({
+  payload,
+}: createPostCommentApiPayload): Promise<
+  ApiResponse<createPostCommentApiResponse>
+> => {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    postId,
+    userId,
+    description,
+    postedAt,
+    lastUpdatedAt,
+  } = payload.comment;
+  return api.post("/api/v1/posts/${postId}/comments", {
```

Comment:
for string interpolation (i.e. using `${}`), don't use `"`, use backtick instead. Otherwise it won't work